velox/common/memory/MemoryUsageTracker.cpp

```diff
@@ -223,29 +223,31 @@ void MemoryUsageTracker::maybeUpdatePeakBytes(int64_t newPeak) {
 }
 
 void MemoryUsageTracker::checkNonNegativeSizes(const char* errmsg) const {
-  // TODO: make these CHECK failures after making usage tracker thread-safe.
-  if (reservationBytes_ < 0) {
-    LOG_EVERY_N(ERROR, kLogEveryN) << "MEMORY: Negagtive reservation bytes "
-                                   << errmsg << " " << reservationBytes_;
-  }
+  VELOX_CHECK_GE(
```

Comment:
Would you deploy custom build to Prestissimo cluster to double check this validation doesn't fail?

Reply:
Yes, I am doing that. Thanks for reminding!